bucket-policy-handlers.go

```diff
@@ -78,14 +78,15 @@ func bucketPolicyActionMatch(action string, statement policyStatement) bool {
 // Verify if given resource matches with policy statement.
 func bucketPolicyResourceMatch(resource string, statement policyStatement) bool {
 	for _, presource := range statement.Resources {
-		matched, err := regexp.MatchString(presource, strings.TrimPrefix(resource, "/"))
-		fatalIf(err, "Invalid pattern, please verify the pattern string.", nil)
-		// For any path matches, we return quickly and the let the caller continue.
-		if matched {
-			return true
+		// the reource rule for object could contain "*" wild card.
+		// TrimSuffix removes the wildcard if it exists and check whether
+		// the requested object can be given access based on the already set bucket policy.
+		// More info: http://docs.aws.amazon.com/AmazonS3/latest/dev/s3-arn-format.html
+		if matched := strings.HasPrefix(resource, strings.TrimSuffix(presource, "*")); !matched {
```

Comment:
But is any other wild card thant `*` is supported by S3 protocols ? If not then there is no need to use Match .